React/Project structure.md

```diff
@@ -0,0 +1,361 @@
+└── components
+    ├── utilities
+    │   └── Meta
+    │       └── Meta.tsx
+    ...
+```
+### UI Components
+
+When adding UI components, you should be able to distinguish them by:
+1. Page scope based components
+2. Components that are shared all accross the app
+3. Large and complex components which contain a lot of stylings, different layouts and child components.
+   
+Lowercase folders are indicators of a page scoped components, shared components and meta components, while PascalCase folders and filenames should be used for components naming.
+
+```
+components
+.
+├── shared
+│   └── WelcomeBox
+│       └── WelcomeBox.tsx
+├── admin
+│   └── AdminBanner
+│       └── AdminBanner.tsx
+└── Gallery
+    ├── Gallery.tsx
+    └── components
+        ├── Arrows.tsx
+        ├── Breadcrumbs.tsx
+        └── ...
+```
+
+#### `components` folder
+
+When adding `components` folder, you basically extracting smaller chunks of your main component that are not going to be used anywhere else, only in that component.
+
+Note: There _should_ be only one level of `components` folder inside the component folder.
+
+## Elements
+
+If you have a lot of style declarations inside you component file (enough to make the file difficult to read), you should create a separated file `elements.ts` where you will store you chakra factories.
+
+Example:
+
+```
+...
+.
+└── WelcomeCard
+    ├── WelcomeCard.tsx
+    └── WelcomeCard.elements.ts
+```
+
+## Different component layouts
+
+Sometimes, you might want to create a molecule specific for mobile and desktop. You might use a tool like [fresnel](https://github.com/artsy/fresnel) for detecting mediaqueries.
+
+In this case, inside a specific molecule, we could add a subfolder `layouts` (not to be confused with actual layout described below) to define how our molecule would look like on specific media query.
+
+```
+.
+└── admin
+    └── UserCard
+        ├── layouts
+        │   ├── UserCard.mobile.tsx
+        │   └── UserCard.desktop.tsx
+        └── UserCard.tsx
+```
+
+For this case, inside `index.tsx` we would have something like this:
+```tsx
+  export const UserCard = () => {
+    return (
+      <Media at="sm">
+        <UserCardMobile />
+      </Media>
+      <Media at="md">
+        <UserCardDesktop />
+      </Media>
+    )
+  }
+```
+
+This is the case **only** for layouts that would add too much complexity when using standard css queries.
+
+
+### Extracting utility functions and hooks
+
+Sometimes, you will have complex functions or effects inside your component that will affect the readability of your component.
+In that case, you should extract them into separated files `utils.ts` and `hooks.ts`.
+
+Main goal of these files is to store functions and hooks that are **specific for that component**, so we could keep our root `hooks` and `utils` folders clean and for global usage purposes only.
+
+Example:
+
+```
+.
+├── ...
+└── AlbumsCarousel
+    ├── AlbumsCarousel.tsx
+    └── AlbumsCarousel.utils.ts
```

Comment:
Just `utils.ts` and `hooks.ts` without component name prefix.
Same reasons as for `elements.ts`

Reply:
why?

Reply:
It's the same reason as for `elements.ts`, answered there ... it's great if this works as expected...